core/src/main/scala/besom/internal/Output.scala

```diff
@@ -77,15 +77,22 @@ trait OutputFactory:
     coll: CC[A]
   )(
     f: A => Output[B]
-  )(using BuildFrom[CC[Output[B]], B, To], Context): Output[To] = sequence(coll.map(f).asInstanceOf[CC[Output[B]]])
+  )(using BuildFrom[CC[Output[B]], B, To], Context): Output[To] = sequence(
+    coll.iterator.map(f).asInstanceOf[CC[Output[B]]]
+  )
   def fail(t: Throwable)(using Context): Output[Nothing] = Output.fail(t)
 trait OutputExtensionsFactory:
   implicit final class OutputSequenceOps[A, CC[X] <: IterableOnce[X], To](coll: CC[Output[A]]):
     def sequence(using BuildFrom[CC[Output[A]], A, To], Context): Output[To] =
       Output.sequence(coll)
   implicit final class OutputTraverseOps[A, CC[X] <: IterableOnce[X]](coll: CC[A]):
     def traverse[B, To](f: A => Output[B])(using BuildFrom[CC[Output[B]], B, To], Context): Output[To] =
-      coll.map(f).asInstanceOf[CC[Output[B]]].sequence
+      coll.iterator.map(f).asInstanceOf[CC[Output[B]]].sequence
+  implicit final class OutputOptionOps[A](output: Output[Option[A]]):
```

Comment:
@lbialy please have a look if those two methods make sense in this context

Reply:
Is there any specific reason why implicit classes are used here instead of scala 3 extension methods?

Reply:
I've just followed the file's style, but actually was surprised as well, CC @lbialy

Reply:
@KacperFKorban had a good reason for `implicit class` here, I think it was related to `import x.*` vs `import x.{*, given}` but that doesn't sound correct in regard to extension methods? Kacper, could you weigh in?

Reply:
This issue is in fact about extension methods.

The thing is that we don't just want to use `extension` methods. That's because extension methods can also be called as regular methods and this litters the scope, since in some cases we only want it to be an extension. That's why a true extension only method is defined as a given with that extension method inside e.g.:
```
given AnyRef with
  extension (i: Int)
    def foo = i + 1
```

This goes back to needing to do `import x.given` in addition to the usual `import x.*`.